lib/engine/round/g_1870/operating.rb

```diff
@@ -7,13 +7,83 @@ module Engine
   module Round
     module G1870
       class Operating < Operating
-        attr_accessor :river_special_tile_lay
+        attr_accessor :river_special_tile_lay, :connection_steps
+        attr_reader :connection_runs
 
         def start_operating
           super
 
           @river_special_tile_lay = nil
         end
+
+        def setup
+          check_connection_runs
+          @connection_steps = []
+
+          super
+        end
+
+        def next_entity!
+          check_connection_runs
+          return if @connection_runs.any?
+
+          super
+        end
+
+        def active_entities
+          return @connection_runs.keys if @connection_runs.any?
+
+          super
+        end
+
+        def check_connection_runs
+          @connection_runs = {}
+          corporations = @game.corporations.dup.sort
+          if current_entity && corporations.any?(current_entity)
+            corporations.unshift(corporations.delete(current_entity))
+          end
+
+          corporations.each do |corporation|
+            next unless (bound = corporation.trains.map(&:distance).max)
+            next unless (destination = @game.destination_hex(corporation))
+            next unless destination.assigned?(corporation)
+
+            home = @game.home_hex(corporation)
+
+            distance = lambda do |hex|
```

Comment:
Is this distance as crow-flies? if so use hex.distance

Reply:
This one is, but hex.distance isn't. I'm changing it regardless. I didn't use it only because I didn't know it existed.